lib/commit-lint.ts

```diff
@@ -0,0 +1,134 @@
+import { IPRCommit } from "./github-glue";
+
+export interface ILintError {
+    checkFailed: boolean;           // true if check failed
+    message: string;
+}
+
+/*
+ * Simple class to drive lint tests on commit messages.
+ */
+export class LintCommit {
+    private failed: boolean;
+    private lines: string[];
+    private patch: IPRCommit;
+    private messages: string[] = [];
+
+    public constructor(patch: IPRCommit) {
+        this.failed = false;
+        this.lines =  patch.message.split("\n");
+        this.patch = patch;
+    }
+
+    /*
+     * Linter method to run checks on the commit message.
+     * @param {IPRCommit} the patch to be checked
+     *
+     * The lintings are in methods called from here.  If
+     * a lint check is too severe to continue, it will throw
+     * an error.
+     */
+    public async lint(): Promise<ILintError | void> {
+
+        // Basic test before all others
+
+        if (await this.commitViable()) {
+            await Promise.all([
+                this.commitMessageLength(),
+                this.lowerCaseAfterPrefix(),
+                this.signedOffBy(),
+                this.moreThanAHyperlink(),
+            ]);
+        }
+
+        if (this.messages.length) {
+            return { checkFailed: this.failed,
+                     message: `There ${this.messages.length > 1 ?
+                     "are issues" : "is an issue"} in commit ${
+                     this.patch.commit}:\n${this.messages.join("\n")}`,
+                };
+        }
+    }
+
+    private addMessage(message: string, failed: boolean = false): void {
+        this.messages.push(message);
+        if (failed) {
+            this.failed = true;
+        }
+    }
```

Comment:
Maybe split this into two methods? For example, keep `addMessage()` as pushing into the array, but then also have `block()` which calls `this.addMessage()` and then sets `this.blocking = true;`? That way, nobody has to remember what that Boolean parameter is about.